proposals/mainnet_migration.md

```diff
@@ -0,0 +1,540 @@
+
+Once Lisk Core v4 processed the snapshot block, it can fetch the whole Lisk Core v3 block history, where this is done from the highest to the lowest height. This allows that each block can be validated by simply checking if its block ID matches with the `previousBlockID` property of the child block, and if the payload matches with the `transactionRoot` property. Thus, almost no protocol rules related to Lisk Core v3 must be implemented which keeps the implementation clean. The Lisk Core v3 history can be fetched from a database created by Lisk Core v3 on the same computer or from the peer-to-peer network. Note that the first Lisk Core v3 block - the snapshot block of the previous migration - poses an exception as this one is either [downloaded from a server](#fetching-snapshot-blocks) or fetched from a database created by Lisk Core v3 on the same computer.
+
+### Fetching Snapshot Blocks
+
+The snapshot block created for this hard fork as well as for the [one used for the previous hard fork](https://lisk.observer/block/abdea6dfe5941922d958d1f1dc1d2f97249723b8852325334e55b73e92584191) are not shared via peer-to-peer due to their large size. Instead, they are either downloaded from a server or must be found locally.
+
+#### Initial Version of Lisk Core v4
+
+The initial version of Lisk Core v4 can not download the snapshot block of this hard fork from a server because it could not validate the block. Instead, it must find the snapshot block locally on the same computer. This requires that the snapshot block must be created using a migrator tool on the same node, which in turn requires that the node runs the latest patched version of Lisk Core v3 until the block at height `HEIGHT_SNAPSHOT` is final.
+
+#### Later Versions of Lisk Core v4
+
+Once the snapshot block is final, a patched version of Lisk Core v4 will be released in which the block ID of the snapshot block is hard coded. This allows to validate a downloaded snapshot block by simply checking its block ID.
+
+#### Downloading from Server
+
+The URL can be configured. By default, the URL points to a server of the Lisk Foundation. The snapshot block of this hard fork is validated by checking that its block ID matches the hard coded block ID in the patched version of Lisk Core v4\. The snapshot block of the previous migration is validated by checking that its block ID matches with the `previousBlockID` property of its child block (see also [here](#getting-the-lisk-core-3-block-history)). Hence, there are no security concerns by downloading them from a non-trusted server. Users can nevertheless configure their node to download it from a server of their choice. The snapshot blocks could be provided by any users. This may also be helpful for situations where the Lisk Foundation server should be down.
```

Comment:
```suggestion
The URL can be configured. By default, the URL points to a server of the Lisk Foundation. The snapshot block for this hard fork is validated by checking that its block ID matches the hard coded block ID in the patched version of Lisk Core v4\. The snapshot block of the previous migration is validated by checking that its block ID matches with the `previousBlockID` property of its child block (see also [here](#getting-the-lisk-core-3-block-history)). Hence, there are no security concerns by downloading them from even a non-trusted server. Users can nevertheless configure their node to download it from a server of their choice. The snapshot blocks could be provided by any user. This may also be helpful in situations where the Lisk Foundation server is down.
```